routes/api/v1/favorites.js

```diff
@@ -144,7 +144,7 @@ router.get('/', function(req, res, next){
         var i;
         const currentForecasts = []
         for (i = 0; i < location.length; i++) {
```

Comment:
if you have time to refactor, I think you can consider using a .map instead here. It takes a little less work. I used this article when doing it that helped me a lot: 
http://cryto.net/~joepie91/blog/2015/05/04/functional-programming-in-javascript-map-filter-reduce/